sycl/plugins/cuda/CMakeLists.txt

```diff
@@ -1,29 +1,11 @@
 message(STATUS "Including the PI API CUDA backend.")
 
- # cannot rely on cmake support for CUDA; it assumes runtime API is being used.
- # we only require the CUDA driver API to be used
- # CUDA_CUDA_LIBRARY variable defines the path to libcuda.so, the CUDA Driver API library.
-
-find_package(CUDA 10.1 REQUIRED)
-
-# Make imported library global to use it within the project.
-add_library(cudadrv SHARED IMPORTED GLOBAL)
-
-if (WIN32)
-  set_target_properties(
-    cudadrv PROPERTIES
-      IMPORTED_IMPLIB               ${CUDA_CUDA_LIBRARY}
-      INTERFACE_INCLUDE_DIRECTORIES ${CUDA_INCLUDE_DIRS}
-  )
-else()
-  set_target_properties(
-    cudadrv PROPERTIES
-      IMPORTED_LOCATION             ${CUDA_CUDA_LIBRARY}
-      INTERFACE_INCLUDE_DIRECTORIES ${CUDA_INCLUDE_DIRS}
-  )
-endif()
-
 if (SYCL_ENABLE_XPTI_TRACING)
+  # cannot rely on cmake support for CUDA; it assumes runtime API is being used.
+  # we only require the CUDA driver API to be used
+  # CUDA_CUDA_LIBRARY variable defines the path to libcuda.so, the CUDA Driver API library.
+  find_package(CUDA 10.1 REQUIRED)
```

Comment:
Why does cmake need 10.1 available when dpc++ is not usually built with 10.1? Can this be removed?:

```
find_package(CUDA 10.1 REQUIRED)
```

Reply:
I think this means that a version of 10.1 or higher is required to build. The Cmake in UR repo is asking for the same version.

Maybe there is an argument that 10.1 is no longer enough? In that case we should decide which version is required and update this cmake and the one in UR (but that is probably out of scope for this PR).

I don't think this can be removed because  `include(FindCUDACupti)` relies on variables set by `find_package(CUDA 10.1 REQUIRED)` such as `CUDA_TOOLKIT_ROOT_DIR`.

Reply:
Was the cuda check moved inside `if (SYCL_ENABLE_XPTI_TRACING)` intentionally? Shouldn't it check there is a cuda version available even when not using XPTI?

Reply:
If the code reached this point, find_package(CUDA) has already been invoked from unified-runtime and the library cudadrv already exists. The only reason to invoke it here is to get the env variables. I moved it inside the if intentionally because it's the only place where it is needed. 

This implicit dependency from unified-runtime CmakeLists is not very intuitive. But PI will be removed soon, so I think there is no need to refactor this too much.